source/vibe/core/core.d

```diff
@@ -1587,3 +1592,10 @@ private template needsMove(T)
 	// FIXME: reverse the condition and only call .move for non-copyable types!
 	enum needsMove = is(typeof(T.init.move));
 }
+
+version(VibeLibasyncDriver) {
+	shared static ~this() {
+		import libasync.threads : destroyAsyncThreads;
+		destroyAsyncThreads(); // destroy threads		
+	}
+}
```

Comment:
> IIUC, this (and code at line 1349) is similar to what was in libasync (at line 1575). Why is the change needed ?
> IMO driver-specific-code should go into core, unless there's no other way.

I get a runtime circular reference error in druntime if I leave it in the driver. This is really the only way I've found, and it's a necessity now.